features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SystemRequirementConfiguration.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.sessionreplay.internal.prerequisite.CpuRequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.DefaultSystemRequirement
+import com.datadog.android.sessionreplay.internal.prerequisite.MemoryRequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.SystemRequirement
+
+/**
+ * This class defines the system requirements necessary to enable the Session Replay feature.
+ *
+ * @param minCPUCoreNumber The minimum CPU core number.
+ * @param minRAMSize The minimum RAM in megabytes.
+ */
+class SystemRequirementConfiguration internal constructor(
+    val minCPUCoreNumber: Int,
+    val minRAMSize: Int,
+    private var systemRequirement: SystemRequirement? = null
+) {
+
+    internal fun runIfMeetRequirements(
+        internalLogger: InternalLogger,
+        runnable: () -> Unit
+    ) {
+        if (systemRequirement == null) {
+            systemRequirement = DefaultSystemRequirement()
+        }
+        systemRequirement?.runIfMeetRequirements(
+            systemRequirementConfiguration = this,
```

Comment:
This API is confusing: `SystemRequirement` is a part of `SystemRequirementConfiguration`, but has a method which takes `systemRequirementConfiguration`. And both have the method with same name. Won't it be easier to detach `SystemRequirement` from configuration and have 2 independent objects, which can be called in `SessionReplay.enable`?

Reply:
it seems we are loosing only very small piece being non-testable due to such move towards `SessionReplay.enable` (basically a single method call verification, I guess?), in exchange getting more clear class structure.